python/cudf/cudf/core/dataframe.py

```diff
@@ -1481,8 +1487,24 @@ def __getitem__(self, arg):
                 nlevels = 1
             elif isinstance(arg, tuple):
                 nlevels = len(arg)
-            if self._data.multiindex is False or nlevels == self._data.nlevels:
+            if (
+                self._data.multiindex is False
+                or nlevels == self._data.nlevels
+                or (
+                    out._data.multiindex is False
+                    and self._data.multiindex is True
+                    and len(out._data.names)
+                    and all(n == "" for n in out._column_names)
+                )
+                or (
+                    out._data.multiindex is True
+                    and self._data.multiindex is True
+                    and len(out._data.names)
+                    and all(n == "" for n in out._column_names[0])
```

Comment:
If `_column_names[0]` is an `int`, wouldn't this raise? I think `_column_names` can be a tuple of hashables

Reply:
It shouldn't raise:
```
In [1]: 1 == ""
Out[1]: False

Reply:
Ah gotcha. I forgot we store MultiIndex column values as tuples - though it's odd that an empty value can still be an empty string. Not to be addressed for this PR